grjobs/analysis/plot_analyses.py

```diff
@@ -0,0 +1,129 @@
+    """
+     Plots cluster dataframe as a scatter plot.  
+
+     Takes as input labelled jobs. Randomly picks x different cluster names and plots them
+     on a scatter plot. 
+         
+    Returns:
+        A labelled scatter plot of job title clusters and their associated cluster label. 
+
+    """ 
+    embedding_df = cluster_job_titles(labelled_jobs)
+    labelled_embedding_df = label_job_title_clusters(embedding_df)
+    
+    fig, ax = plt.subplots(figsize=(20, 10))
+    clustered = labelled_embedding_df.loc[labelled_embedding_df.labels != -1, :]
+    
+    x = clustered.x.tolist()
+    y = clustered.y.tolist()
+    
+    random_cluster_names = random.sample(list(set(clustered['cluster_name'].astype(str).tolist())), random_cluster_no)
+
+    cluster_labels = []
+    x_labels = []
+    y_labels = []
+
+    for cluster_name in random_cluster_names:
+        x_label = clustered[clustered['cluster_name'].astype(str) == cluster_name]['x'].tolist()
+        y_label = clustered[clustered['cluster_name'].astype(str) == cluster_name]['y'].tolist()
+        cluster_labels.append(cluster_name.replace('[', '').replace(']', '').replace("'", ''))
+        x_labels.append(random.choice(x_label))
+        y_labels.append(random.choice(y_label))
+    
+    plt.scatter(x, 
+            y, 
+            c=clustered.labels, 
+            s=1, 
+            cmap='hsv_r')
+
+    for i, txt in enumerate(cluster_labels):
+        plt.annotate(txt, 
+                 (x_labels[i], 
+                  y_labels[i]),
+                fontsize=15,
+                fontproperties=prop,
+                color='black')
+
+    plt.axis('off')
+    ax.set_title('Job Title Clusters in Green Industries', 
+             fontproperties=prop,
+            size=25,
+            color="black")
+    color = 'viridis'
+
+    return plt
+
+def plot_green_locations(labelled_jobs) -> plt:
+    """
+     Plots percentage of job adverts in green industries nuts 2 code. 
+         
+    Returns:
+        An ordered bar chart of the percentage of 'green' jobs by UK nuts 2 code.
+
+    """ 
+    green_job_locations = []
+    for job in labelled_jobs:
+        if 'location' in job['features'].keys():
+            green_job_locations.append(job['features']['location']['nuts_2_code'])
+
+    all_locations = []
+    green_ids = get_green_ids(labelled_jobs)
+    for job in get_db_job_ads(limit = None, return_features = True):
+        if job['id'] not in green_ids and 'location' in job['features'].keys():
+            all_locations.append(job['features']['location']['nuts_2_code'])
+
+    green_locations_df = pd.DataFrame(Counter(green_job_locations), index = ['green_count']).T
+    green_locations_df['NUTS2_CODE'] = green_locations_df.index
+    all_locations_df =  pd.DataFrame(Counter(all_locations), index = ['all_count']).T
+    all_locations_df['NUTS2_CODE'] = all_locations_df.index
+
+    locations_df = green_locations_df.merge(all_locations_df, on = 'NUTS2_CODE')
```

Comment:
.merge does an inner join by default - is there a risk here that there are regions with no green jobs that don't appear in locations_df since they don't feature in green_locations_df?

edit: this is actually fine because you only plot the top 5 - but might need tweaking if you ever wanted to plot the bottom 5 instead